prov/usnic/src/usdf_eq.c

```diff
@@ -389,13 +317,13 @@ usdf_eq_write_fd(struct fid_eq *feq, uint32_t event, const void *buf,
 	ret = usdf_eq_write_event(eq, event, buf, len, flags);
 
 	/* If successful, post to eventfd */
-	if (ret >= 0) {
-		val = 1;
+	if (ret >= 0 && eq->eq_wait_obj == FI_WAIT_FD) {
 		n = write(eq->eq_fd, &val, sizeof(val));
-		if (n != sizeof(val)) {
+
+		/* TODO: If the write call fails, then roll back the EQ entry.
+		 */
+		if (n != sizeof(val))
```

Comment:
What's the expected behavior if enqueuing something succeeds, but notifying the FD fails? Do I roll back the enqueue and go into an errored state?